File: 165-compare-version-numbers/165-compare-version-numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        li = version1.split('.')
        li2 = version2.split('.')
        
        if(len(li) > len(li2)):
            diff = len(li) - len(li2)
            for i in range (diff):
                li2.append('0')
                
        elif(len(li2) > len(li)):
            diff = len(li2) - len(li)
            for i in range (diff):
                li.append('0')
                
        for i in range (len(li)):
            if int(li[i]) > int(li2[i]):
                return 1
            elif int(li[i]) < int(li2[i]):
                return -1
        return 0
        

File: 165-compare-version-numbers/165-compare-version-numbers.cpp
class Solution {
public:
    int compareVersion(string v1, string v2) {
        int n = v1.size() , m = v2.size() , i = 0 , j = 0;
        while(i < n or j < m)
        {
            int n1 = 0 , n2 = 0;
            while(i < n and v1[i] != '.')
            {
                n1 = n1 * 10 + (v1[i] - '0'); 
                i++;
            }
            while(j < m and v2[j] != '.')
            {
                n2 = n2 * 10 + (v2[j] - '0');
                j++;
            }
            
            if(n1 > n2)
                return 1;
            else if(n1 < n2)
                return -1;
            i++;
            j++;
        }
        return 0;
    }
};